common/src/main/java/com/microsoft/identity/common/adal/internal/util/StringExtensions.java

```diff
@@ -245,6 +248,98 @@ public static boolean hasPrefixInHeader(final String value, final String prefix)
      * @return String
      */
     public static String base64UrlEncodeToString(final String message) {
-        return Base64.encodeToString(message.getBytes(Charset.forName(ENCODING_UTF8)), Base64.URL_SAFE | Base64.NO_WRAP);
+        return Base64.encodeToString(message.getBytes(Charset.forName(AuthenticationConstants.ENCODING_UTF8)), Base64.URL_SAFE | Base64.NO_WRAP);
+    }
+
+    /**
+     * Convert the given set of scopes into the string with the provided delimiter.
+     *
+     * @param inputSet  The Set of scopes to convert.
+     * @param delimiter The delimiter used to construct the scopes in the format of String.
+     * @return The converted scopes in the format of String.
+     */
+    public static String convertSetToString(final Set<String> inputSet, final String delimiter) {
```

Comment:
Should these be part of a generic string utility class?  Or should we have class focused on handling scopes?